src/core/task/index.ts

```diff
@@ -3003,25 +3025,25 @@ export class Task {
 				const toolUseBlocks = this.toolUseHandler.getAllFinalizedToolUses()
 
 				// Build content array with thinking blocks, text (if any), and tool use blocks
-				const assistantContent: Array<
-					| Anthropic.Messages.RedactedThinkingBlock
-					| Anthropic.Messages.ThinkingBlock
-					| Anthropic.Messages.TextBlockParam
-					| Anthropic.ToolUseBlockParam
-				> = [
+				const assistantContent: Array<ClineAssistantContent> = [
 					// This is critical for maintaining the model's reasoning flow and conversation integrity.
 					// "When providing thinking blocks, the entire sequence of consecutive thinking blocks must match the outputs generated by the model during the original request; you cannot rearrange or modify the sequence of these blocks." The signature_delta is used to verify that the thinking was generated by Claude, and the thinking blocks will be ignored if it's incorrect or missing.
 					// https://docs.claude.com/en/docs/build-with-claude/extended-thinking#preserving-thinking-blocks
-					...antThinkingContent,
+					...redactedThinkingContent,
 				]
-
+				if (reasoningMessage.trim().length > 0) {
+					assistantContent.push({
+						type: "thinking",
+						thinking: reasoningMessage,
+						signature: reasoningSignature,
+					})
+				}
 				// Only add text block if there's actual text (not just tool XML)
 				if (assistantTextOnly.trim().length > 0) {
 					assistantContent.push({
 						type: "text",
 						text: assistantTextOnly,
 						// reasoning_details only exists for cline/openrouter providers
-						// @ts-ignore-next-line
```

Comment:
we used to store reasoning_details as part of text content in a field that is not known to Anthropic SDK using `@ts-ignore-next-line` so the downstream callers are unaware of this field and would include it accidentally when the api is not supported. Having a standardized interface allows us to be typesafe and ensure this doesn't happen in the future.